mozdownload/scraper.py

```diff
@@ -135,7 +135,7 @@ def _retry_check_404(self, func,
                                  requests.exceptions.RequestException))
         try:
             self._retry(func, **retry_kwargs)
-        except requests.exceptions.RequestException as exc:
+        except requests.exceptions.HTTPError as exc:
```

Comment:
This change looks fine to me. Lets see which ambiguous exception has been raised by requests in those rare occurrences.